src/Hazelcast.Net/Serialization/Compact/CompactSerializationSerializer.cs

```diff
@@ -83,12 +86,18 @@ public void Dispose()
             // note: ISchemas is IDisposable but is owned by the global SerializationService
         }
 
+        /// <inheritdoc cref="IStreamSerializer{T}.Read"/>
         public object Read(IObjectDataInput input)
-            => ReadObject(input.MustBe<ObjectDataInput>(nameof(input)));
+            // NOTE: we *have* to support this method because of IStreamSerializer<object>,
+            // but then we have to pass typeof(object) as the expected object type, which
+            // means that we cannot specify that we want a IGenericRecord. We'll get what
```

Comment:
Can you explain more? Why cannot we specifiy that we want a generic record?

Reply:
At top level, `CompactSerializationSerializer` can serialize every kind of objects and therefore implements `IStreamSerializer<object>` so it *has* to implement that method. But that comment is misleading, it's kind of a leftover. We don't *have* to specify that we want a generic record and in fact we may get a plainly valid other object. Getting rid of the comment.

Reply:
ok